pkg/keyctl/keyring_test.go

```diff
@@ -0,0 +1,35 @@
+// +build linux
+
+package keyctl
+
+import (
+	"crypto/rand"
+	"testing"
+)
+
+func TestSessionKeyring(t *testing.T) {
+
+	token := make([]byte, 20)
+	rand.Read(token)
+
+	testname := "testname"
+	keyring, err := SessionKeyring()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = keyring.Add(testname, token)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key, err := keyring.Search(testname)
+	if err != nil {
+		t.Fatal(err)
+	}
+	data, err := key.Get()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(data) != string(token) {
+		t.Errorf("Expected data %v, but get %v", token, data)
+	}
+}
```

Comment:
Parts of the API are not yet tested (e.g., Unlink, Link, ID). Let's add some more tests to test them in separate functions to increase the test coverage.